src/main/java/org/spongepowered/api/event/entity/living/player/PlayerChangeSettingsEvent.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.entity.living.player;
+
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.text.chat.ChatTypes;
+
+import java.util.Locale;
+
+/**
+ * Fired when A {@link Player} changes one or more of the following settings:
+ *
+ * <ul>
+ *  <li>Locale</li>
+ *  <li>View distance</li>
+ *  <li>Chat visibility</li>
+ *  <li>Chat colors</li>
+ *  <li>Displayed skin parts</li>
+ * </ul>
+ */
+public interface PlayerChangeSettingsEvent extends TargetPlayerEvent {
```

Comment:
imo I would name this `PlayerChangeClientSettingsEvent` just to make it more clear what settings they're changing.